### PR TITLE
[REF] stock: _action_assign

### PR DESCRIPTION
**Pull request by sle-odoo**
`_update_reserved_quantity`

opw-1835186
opw-1835946

Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/stock/models/stock_move.py [hunk 1]
```diff
@@ -815,12 +824,22 @@ def _update_reserved_quantity(self, need, available_quantity, location_id, lot_i
 
         taken_quantity = min(available_quantity, need)
 
+        # `taken_quantity` is in the quants unit of measure. There's a possibility that the move's
+        # unit of measure won't be respected if we blindly reserve this quantity, a common usecase
+        # is if the move's unit of measure do not allow fractionnal reservation. We chose to
```
Comment (jco-odoo) [inline addons/stock/models/stock_move.py]: does not / fractional  / maybe talk about roundings not respecting....    (if the roudings of the uoms don't allow...)

**Comment by sle-odoo** [comment]
> Decimal Precision for uom roundings

it is only used to compute `quantity_done` and updating `product_qty` on a move that has been backordered. they are technical fields.

> correct roundings on their uoms

what's wrong with setting 1.0 as rounding on the dozen?

**Comment by jco-odoo** [comment]
I think it is fundamentally wrong.  The UoMs should have a consistent
rounding towards the UoM of reference.  We can also avoid the user to do
inconsistent uom roundings in configuring his uoms.

File: addons/stock/models/stock_move.py [hunk 2]
```diff
@@ -247,10 +247,19 @@ def _get_move_lines(self):
...
+        """
         for move in self:
+            quantity_done = 0
             for move_line in move._get_move_lines():
-                # Transform the move_line quantity_done into the move uom.
-                move.quantity_done += move_line.product_uom_id._compute_quantity(move_line.qty_done, move.product_uom)
+                quantity_done += move_line.product_uom_id._compute_quantity(move_line.qty_done, move.product_uom, round=False)
+            move.quantity_done = float_round(quantity_done, precision_digits=self.env['decimal.precision'].precision_get('Product Unit of Measure'))
 
     def _quantity_done_set(self):
```
Comment (sle-odoo) [inline addons/stock/models/stock_move.py]: issue here: with this patch, it is possible to set 0.5 even if the move's uom has a rounding of 1.0

**Pull request merged**